### PR TITLE
Structured logging

### PR DESCRIPTION
**Pull request by stefanceriu**
This PR adops the new RustSDK structured logging APIs directly through MXLog and cleans up the associated components
It also cleans up timeline diffind logs and promotes them to Info

NB There will be another SDK update linked to this PR as soon as available which tweaks the logger configuration Rust side to include location information

### CODE REVIEW

File: ElementX/Sources/Other/Logging/MXLogger.swift [hunk 1]
```diff
@@ -39,12 +39,14 @@ class MXLogger {
     ///   - redirectToFiles: `true` to enable the redirection.
     ///   - numberOfFiles: number of files to keep (default is 10).
     ///   - sizeLimit: size limit of log files in bytes. 0 means no limitation, the default value for other methods
```
Comment (pixlwave) [inline ElementX/Sources/Other/Logging/MXLogger.swift:41]:
```suggestion
    ///   - maxLogFileCount: number of files to keep (default is 10).
    ///   - logFileSizeLimit: size limit of log files in bytes. 0 means no limitation, the default value for other methods
```

File: UnitTests/Sources/LoggingTests.swift [hunk 2]
```diff
@@ -73,14 +70,13 @@ class LoggingTests: XCTestCase {
         // Given a fresh launch with no logs.
         XCTAssertTrue(MXLogger.logFiles.isEmpty)
         
+        MXLog.configure()
```
Comment (pixlwave) [inline UnitTests/Sources/LoggingTests.swift:73]: How come the extra call to `configure()` here?

File: UnitTests/Sources/LoggingTests.swift [hunk 3]
```diff
@@ -95,14 +91,13 @@ class LoggingTests: XCTestCase {
...
```
Comment (pixlwave) [inline UnitTests/Sources/LoggingTests.swift:94]: And here.

File: ElementX/Sources/Other/Logging/RustTracing.swift [hunk 4]
```diff
@@ -22,11 +22,24 @@ import MatrixRustSDK
...
+    static var debug = TracingConfiguration(overrides: [.matrix_sdk_crypto: .info,
+                                                        .matrix_sdk_http_client: .info,
+                                                        .matrix_sdk_sliding_sync: .info,
+                                                        .matrix_sdk_base_sliding_sync: .info])
+    
     static func custom(overrides: [Target: LogLevel]) -> TracingConfiguration {
         TracingConfiguration(overrides: overrides)
     }
     
+    static func custom(logLevel: LogLevel) -> TracingConfiguration {
```
Comment (pixlwave) [inline ElementX/Sources/Other/Logging/RustTracing.swift:35]: Could we add a brief doc to both of these just to explain the difference without having to look at the code?

File: ElementX/Sources/Other/Logging/MXLog.swift [hunk 5]
```diff
@@ -15,100 +15,133 @@
...
-    public static func configure(_ configuration: MXLogConfiguration) {
-        configureLogger(logger, withConfiguration: configuration)
+enum MXLog {
+    private enum Constants {
+        static let target = "elementx"
+        
+        // Avoid redirecting NSLogs to files if we are attached to a debugger.
+        static let redirectToFiles = isatty(STDERR_FILENO) == 0
+        
+        /// the maximum number of log files to use before rolling. `50` by default.
```
Comment (pixlwave) [inline ElementX/Sources/Other/Logging/MXLog.swift:31]:
```suggestion
        /// the maximum number of log files to use before rolling. `10` by default.
```

**Review by pixlwave — APPROVED** [review]
Looks good to me.

File: UnitTests/Sources/LoggingTests.swift (hunk 2 above)
Comment (stefanceriu) [reply UnitTests/Sources/LoggingTests.swift:73]: I wish I could remember but no idea really ¯\_(ツ)_/¯ Got rid of both of them, good catch 👏

**Pull request merged**